pkg/kvstore/etcd.go

```diff
@@ -593,7 +612,11 @@ func connectEtcdClient(ctx context.Context, config *client.Config, cfgPath strin
 	ec.controllers.UpdateController("kvstore-etcd-lock-session-renew",
 		controller.ControllerParams{
 			DoFunc: func(ctx context.Context) error {
-				return ec.renewLockSession()
+				// Same context logic as above.
```

Comment:
Might be worth copy and pasting here - 'above' might change in the future.

Reply:
My plan after the bugfix would be to refactor this code so we don't have to be copy+pasting :-)